feat: 주문 구현

Manhye opened the pull request:
## 작업내용
주문 부분을 작업하였습니다.

## 팀원에게 전할 말
에러부분은 나중에 정리되면 수정하겠습니다.

Manhye committed:
feat: 주문 구현

dungbik commented on src/main/java/com/example/dishpatch/infra/db/pointHistory/repository/PointHistoryRepository.java:
```diff
@@ -1,8 +1,19 @@
 public interface PointHistoryRepository extends JpaRepository<PointHistory, Long> {
+
+	@Query("SELECT SUM(p.remain) FROM PointHistory p WHERE p.user.id = :userId AND p.pointUsed = :pointUsed")
+	Integer sumRemainByUserIdAndPointUsed(@Param("userId") Long userId, @Param("pointUsed") PointUsed pointUsed);
```
만료된 포인트 이력은 더해지지 않게 해야하지 않나요?

Manhye replied:
public Integer getRemainingPoint(Long userId) {

		return pointHistoryRepository.sumRemainByUserIdAndPointUsed(
			userId, PointUsed.UNUSED);
	}
	이게 여기 PointHistoryService에서 이용되는데 UNUSED만 받아와서 더해지게 설계했습니다.

dungbik replied:
만료되도 DeletedDate만 생기고 PointUsed가 안바뀌지 않나요??

Manhye replied:
아 아직 만료되는 기능을 따로 만들어두지 않았습니다. 그건 주문 처리 부분에서 처리하는것은 아니지 않나요??

dungbik replied:
포인트 계산할때 만료된건 포함을 안시켜야하는데 지금 상태로는 만료 체크가 없어서 만료된 것도 더해집니다!

dungbik commented on src/main/java/com/example/dishpatch/domain/pointHistory/service/PointHistoryService.java:
```diff
@@ -0,0 +1,49 @@
+
+		int index = 0;
+
+		while (point > 0) {
```
update가 건바이건으로 일어나기 때문에 이부분은 벌크로 작업하는게 좋을 것 같습니다

dungbik commented on src/main/java/com/example/dishpatch/domain/order/service/OrderService.java:
```diff
@@ -0,0 +1,116 @@
+			return totalPrice;
+		}
+
+		if (coupon.getCoupontype() == CouponType.A) {
```
쿠폰 타입 A, B 보다 좀더 직관적인 이름을 쓰면 좋을 것 같습니다

dungbik commented:
복잡해서 보기도 힘드네요 고생 많으셨습니다...

somin-jeong commented on src/main/java/com/example/dishpatch/domain/coupon/service/CouponService.java:
```diff
@@ -0,0 +1,28 @@
+	public Coupon getCoupon(Long couponId) {
+
+		Coupon coupon = couponRepository.findById(couponId)
+			.orElseThrow(() -> new RuntimeException("쿠폰이 존재하지 않습니다.")
```
에러 코드 사용해주시면 좋을 것 같습니다!

somin-jeong reviewed (dismissed):
고생하셨습니다

Manhye committed:
B
Merge branch 'dev' of https://github.com/somin-jeong/DishPatch into feat/order-order
merge

Manhye committed:
Feat: 주문 현황 업데이트 구현

Manhye committed:
Feat: 주문 현황 업데이트 구현
